0003-lightning-as-alpha-ledger.adoc

```diff
@@ -0,0 +1,279 @@
+7. Bob gets notified about this, learns the secret and can now settle the LND invoice by invoking the LND command `settleInvoice` (https://github.com/lightningnetwork/lnd/blob/aa1cd04dbf07a9195d5ada752f383988d8d01fa7/cmd/lncli/invoicesrpc_active.go#L53[more details]).
+
+The trade is complete.
+
+== Spike Outcome
+
+=== Lightning Network: a new ledger
+Similiar to the ledger definitions for https://github.com/comit-network/RFCs/blob/master/RFC-004-Bitcoin.md[Bitcoin] and https://github.com/comit-network/RFCs/blob/master/RFC-006-Ethereum.md[Ethereum] we need to handle the Lightning Network differently.
+This is required because the comit-node and btsieve need to perform different actions accordingly.
+We are always talking about Ledgers and Assets, (e.g. _Bitcoin_ Asset on the _Bitcoin_ Ledger, _Ether_ Asset on the _Ethereum_ Ledger, _Erc20_ on the _Ethereum_ Ledger, ...), Hence,
+if we follow this approach, for supporting LN (through LND) we will need to introduce a new pair of Ledger and Asset:
+
+* Ledger: the **Lightning Network**. _Ledgers_ are used as _settlement layers_ for our HTLCs. In the case of LND, this layer is the Lightning Network.
+* Asset: **Bitcoin**. Since LN is a layer-2 network on top of Bitcoin, the asset should also be Bitcoin.
+
+=== Dealing with timeouts
+When calling `addholdinvoice` an https://github.com/lightningnetwork/lnd/blob/aa1cd04dbf07a9195d5ada752f383988d8d01fa7/cmd/lncli/invoicesrpc_active.go#L142[`expiry`] (in seconds) can be given. However, this expiry is only related to the invoice itself and has nothing to do with the expiry times on the HTLCs later on.
+However, timeouts on the HTLCs are relative and will get accumulated across the payment path, which makes it harder to predict.
+Hence, Alice will need to know the path in advance before sending a swap request to Bob as she won't (or should not) be able to change neither `alpha_expiry` nor `beta_expiry` after the swap request.
+
+Timeouts on the HTLCs are configured globally for the payee node and can optionally be defined when paying by the payer node (`--final_cltv_delta=T`).
+
+
+IMPORTANT: TODO: we should investigate this topic more in detail.
+
+=== Responsibilities
+
+A main goal of COMIT is to keep the autonomy to the user and let him/her decide when to deploy a HTLC, redeem or refund a HTLC, etc.
+If a trade involves LN using LND we can approach these things differently:
+
+
+* Action
+    ** Create hold invoice
+* Responsibility
+    ** LND
+* Invoker
+    ** User, comit-i, comit-node
+* Description
+    ** `addholdinvoice` is available as a RPC command or through the LND CLI. Although dealing with this is rather cumbersome, to keep the autonomy with the user, and to not introduce LND dependency into the comit-node, we this should be possible through comit-i.
+* Conclusion:
+    ** comit-i needs LND support. However, in order to do this, we will need to introduce a new action which is meant to be executed prior accepting a swap request:
```

Comment:
Meh, I disagree. If things have to be done atomically (sending accept + adding the invoice) then you should really build it in a way so that the user does _NOT_ have a choice to break this invariant. For me, such atomicity is not guaranteed just because comit-i exposes it as a single action. From that, I would conclude that the comit-node should perform the `addholdinvoice` action by talking to LND. Thinking ahead (comit-node speaking L1 and possibly L2 protocols directly), this seems to be a path that makes such a transition much easier.

Reply:
I think it's a valid debate we discussed yesterday. Do you expose the invoice creation as an action in the HTTP API or do you have `comit_node` do it for you.

But then, if `comit_node` is the one creating the hold invoice, why not being the one doing the payment?

And if you have `comit_node` connected to lnd, then why not do the watching directly from lnd instead of having btsieve connected to local lnd?

I think the architectural point of what service talks to which service may deserve a live discussion.